tests/test_kex.py

```diff
@@ -515,7 +527,12 @@ def test_15_kex_c25519_client(self):
         transport = FakeTransport()
         transport.server_mode = False
         kex = KexCurve25519(transport)
-        kex.start_kex()
+        try:
+            kex.start_kex()
+        except UnsupportedCryptographyVersionError:
+            return pytest.skip(
+                "cryptography version is too old to use x25519"
```

Comment:
with your latest changes, is it possible to remove this check, because `KexCurve25519.is_supported()` now returns `False`?

Reply:
The `is_supported()` method checks only for underlying X25519 support and `Encoding.Raw` + `PublicFormat.Raw`. This error is thrown when there's no `from_private_bytes`/`_from_private_bytes`. We only need `from_private_bytes()` for unit tests, which is why this check is kept out of `KexCurve25519.is_supported()` and this catch is preserved here.

Reply:
Right, but `X25519.from_private_bytes()` was introduced at the same time as `Encoding.Raw` (in cryptography-2.5) ... so we can take a bit of a shortcut and just assume that if KexCurve25519 is supported, we must have Encoding.Raw, _and_ `from_private_bytes()` (without `_`).

Just allows a minor simplification, not something I care too strongly about.